### DISCUSSION TITLE
zsh history goes wonky when scrolling up

### DISCUSSION THREAD

**Discussion by wgroenewald** [ohmyzsh/ohmyzsh#10172] [Help] [answered]
When in terminal and pressing the up arrow to go through my zsh history the line will often get messed up. This is especially true when the history is multiline or has special characters in it.

It's difficult to provide and example, but if my history is:
```
1 echo hello world
2 export foo=bar
3 cat /my/file.txt
```
Then pushing up will give me first `cat /my/file.txt` then `export foo=bar` and then it may break and do something like `execho hello world`. Taking the first couple of letters from the previous command and appending it. Going back down then will stay broken. It looks to be more when the commands have multiline, or certain characters - though I'm not sure.

It seems this is only an issue on some themes. Tested a few

Themes with break my history scrolling:
- robbyrussel
- apple
- bira
- arrow

Themes which seem to work fine:
- dallas
- alanpeabody
- rgm

**Comment by mcornella** [comment, marked as answer]
This is probably a locale issue, see [this FAQ answer](https://github.com/ohmyzsh/ohmyzsh/wiki/FAQ#i-see-duplicate-typed-characters-after-i-complete-a-command).
